File: CS/Scaffolding.DetailCollections/DepartmentContextDataModel/DepartmentContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.DetailCollections.Common.Utils;
using Scaffolding.DetailCollections.Common.DataModel;
using Scaffolding.DetailCollections.Common.DataModel.EntityFramework;
using Scaffolding.DetailCollections.Model;

namespace Scaffolding.DetailCollections.DepartmentContextDataModel {
    /// <summary>
    /// A DepartmentContextUnitOfWork instance that represents the run-time implementation of the IDepartmentContextUnitOfWork interface.
    /// </summary>
    public class DepartmentContextUnitOfWork : DbUnitOfWork<DepartmentContext>, IDepartmentContextUnitOfWork {

        public DepartmentContextUnitOfWork(Func<DepartmentContext> contextFactory)
            : base(contextFactory) {
        }

        IRepository<Course, int> IDepartmentContextUnitOfWork.Courses {
            get { return GetRepository(x => x.Set<Course>(), x => x.CourseID); }
        }

        IRepository<Department, int> IDepartmentContextUnitOfWork.Departments {
            get { return GetRepository(x => x.Set<Department>(), x => x.DepartmentID); }
        }

        IRepository<Employee, int> IDepartmentContextUnitOfWork.Employees {
            get { return GetRepository(x => x.Set<Employee>(), x => x.EmployeeID); }
        }
    }
}

File: VB/Scaffolding.DetailCollections/DepartmentContextDataModel/DepartmentContextUnitOfWork.vb
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.DetailCollections.Common.Utils
Imports Scaffolding.DetailCollections.Common.DataModel
Imports Scaffolding.DetailCollections.Common.DataModel.EntityFramework
Imports Scaffolding.DetailCollections.Model

Namespace Scaffolding.DetailCollections.DepartmentContextDataModel
    ''' <summary>
    ''' A DepartmentContextUnitOfWork instance that represents the run-time implementation of the IDepartmentContextUnitOfWork interface.
    ''' </summary>
    Public Class DepartmentContextUnitOfWork
        Inherits DbUnitOfWork(Of DepartmentContext)
        Implements IDepartmentContextUnitOfWork

        Public Sub New(ByVal contextFactory As Func(Of DepartmentContext))
            MyBase.New(contextFactory)
        End Sub

        Private ReadOnly Property IDepartmentContextUnitOfWork_Courses() As IRepository(Of Course, Integer) Implements IDepartmentContextUnitOfWork.Courses
            Get
                Return GetRepository(Function(x) x.Set(Of Course)(), Function(x) x.CourseID)
            End Get
        End Property

        Private ReadOnly Property IDepartmentContextUnitOfWork_Departments() As IRepository(Of Department, Integer) Implements IDepartmentContextUnitOfWork.Departments
            Get
                Return GetRepository(Function(x) x.Set(Of Department)(), Function(x) x.DepartmentID)
            End Get
        End Property

        Private ReadOnly Property IDepartmentContextUnitOfWork_Employees() As IRepository(Of Employee, Integer) Implements IDepartmentContextUnitOfWork.Employees
            Get
                Return GetRepository(Function(x) x.Set(Of Employee)(), Function(x) x.EmployeeID)
            End Get
        End Property
    End Class
End Namespace
